client/state/data-getters/index.js

```diff
@@ -176,6 +176,20 @@ export const requestGutenbergDemoContent = () =>
 		{ fromApi: () => data => [ [ 'gutenberg-demo-content', data ] ] }
 	);
 
+export const requestSelectedEditor = siteId =>
+	requestHttpData(
+		'selected-editor',
+		http(
+			{
+				path: `/sites/${ siteId }/gutenberg`,
+				method: 'GET',
+				apiNamespace: 'wpcom/v2',
+			},
+			{}
```

Comment:
This is as far as I got @dmsnell ; I'm thinking I'm missing `setSelectedEditor` here (although I don't see how to pass its params in the way it would need)?

Reply:
```js
( siteId, editor ) => http( {
  path: …,
  method: …,
  onSuccess: setEditor( siteId, editor )
} )
```